### DISCUSSION TITLE
How to run "createItems" from MenuButton after the constructor of MenuButton.

### DISCUSSION THREAD

**Discussion by Myroslav-Bilous** [videojs/video.js#8661] [Q&A]
Hi. I have a class AudioMenuButton extended from MenuButton, and my method createItems use data from AudioMenuButton like this.settings. But when it is run this.settings is undefined because createItems runs in cunstructor of parent class MenuButton before passing data.
Does anyone know how I can solve this problem?

**Comment by mister-ben** [comment]
Calling `this.update()` might work for you.
